### PR TITLE
Amending uploadform and etc.

### PR DESCRIPTION
**Pull request by kazutan1230**
### What does this PR do?

- app/download/page.tsx
- components/jsonView.tsx
- components/useFormData.tsx
Adding a download page mock, thanks to @tanePON 

- components/imageInput.tsx
- components/imagePreview.tsx
- components/uploadFormHook.tsx
I deleted the imagePreview.tsx and merged the function of imagePreview to uploadFormHook.tsx,
because it is easy to register images into FormData.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

Github workflow checks have been passed.
I cheked at localhost and browser .

### CODE REVIEW

File: components/uploadFormHook.tsx [hunk 1]
```diff
@@ -1,39 +1,79 @@
...
 export const UploadFormHook: FC = () => {
-  const { handleSubmit, register } = useForm<FormData>()
+  const { handleSubmit, register, setValue } = useForm<FormData>()
+
+  // 添付画像を状態管理
+  const [images, setImages] = React.useState<{
+    file: File | null
+    name: string
+    source: string
+  }>({ file: null, name: '', source: '' })
```
Comment (Marukome0743) [inline components/uploadFormHook.tsx:37]:
```suggestion
type ImageData = {
  file: File
  name: string
  source: string
}

   // 添付画像を状態管理
  const [images, setImages] = React.useState<ImageData>({ file: new File([], ""), name: '', source: '' })
```

File: components/uploadFormHook.tsx [hunk 2]
```diff
@@ -1,39 +1,79 @@
...
+  }
+
+  // キャンセルボタンの処理
+  const handleClickCancelButton = () => {
+    setImages({
+      ...images,
+      file: null,
+      name: '',
+      source: '',
+    })
```
Comment (Marukome0743) [inline components/uploadFormHook.tsx:69]:
```suggestion
  // キャンセルボタンの処理
  const handleClickCancelButton = () => {
    setImages({
      ...images,
      file: new File([], ""),
      name: '',
      source: '',
    })
```

**Review by Marukome0743 — CHANGES_REQUESTED** [review]

File: components/uploadFormHook.tsx [hunk 3]
```diff
@@ -112,16 +158,51 @@ export const UploadFormHook: FC = () => {
...
+                画像をアップロードする
+              </span>
+            )}
+            <ImageInput
+              onChange={handleImageChange}
+              id="image"
+              required={true}
+            />
+          </label>
+          <div>
```
Comment (Marukome0743) [inline components/uploadFormHook.tsx:187]: Remove unused `<div>`.

**Review by Marukome0743 — CHANGES_REQUESTED** [review]

File: app/download/page.tsx [hunk 4]
```diff
@@ -1,112 +1,116 @@
...
-              -&gt;
-            </span>
-          </h2>
-          <p className={'m-0 max-w-[30ch] text-sm opacity-50'}>
-            Instantly deploy your Next.js site to a shareable URL with Vercel.
-          </p>
-        </a>
+  return (
+    <main className="flex min-h-screen flex-col items-center justify-between p-24">
+      <div className="hoge">
```
[the remark this answers is not in the document]
Comment (kazutan1230) [reply app/download/page.tsx:33]: I'll remove it.

File: app/download/page.tsx [hunk 5]
```diff
@@ -1,112 +1,116 @@
...
+              <th className="px-6 py-4 font-medium text-gray-900">
+                Helen Howard
+              </th>
+              <td className="px-6 py-4">Nov.4 2022</td>
+              <td className="px-6 py-4">helen@sailboatui.com</td>
+              <td className="flex justify-end gap-4 px-6 py-4 font-medium">
+                <a href="hoge">Delete</a>
+                <a href="hoge" className="text-primary-700">
+                  Edit
+                </a>
```
[the remark this answers is not in the document]
Comment (kazutan1230) [reply app/download/page.tsx:109]: I'll change from<a> to <button>.

File: components/jsonView.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  if (loading) {
+    return <p>Loading...</p>
+  }
+  const localFormData = formdata
+  if (localFormData === undefined) {
+    return <p>No Json Data</p>
+  }
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (kazutan1230) [reply components/jsonView.tsx:17]: I'll remove <div>.

File: components/uploadFormHook.tsx (hunk 3 above)
Comment (kazutan1230) [reply components/uploadFormHook.tsx:187]: I'll remove `<div>`.

**Review by Marukome0743 — APPROVED** [review]

**Review by HaradaTakayuki — APPROVED** [review]
確認しました

**Pull request merged**